### PR TITLE
added logging for code lens additions

### PR DESCRIPTION
**Pull request by amunger**
For #8910

troubleshooting flakey code lens test where the code lenses are never added to the document

### CODE REVIEW

File: src/client/datascience/editor-integration/codelensprovider.ts [hunk 1]
```diff
@@ -161,7 +162,7 @@ export class DataScienceCodeLensProvider implements IDataScienceCodeLensProvider
             return codeWatcher.getCodeLenses();
         }
 
-        // Create a new watcher for this file
+        traceInfoIfCI(`Creating a new watcher for document ${document.uri}`);
         const newCodeWatcher = this.createNewCodeWatcher(document);
```
Comment (amunger) [inline src/client/datascience/editor-integration/codelensprovider.ts:166]: When does this case happen? (the documentManager doesn't contain a document with the matching URI)

Comment (rchiodo) [reply src/client/datascience/editor-integration/codelensprovider.ts:166]: Document manager doesn't? That seems odd. 

I might have expected the version numbers to not match but the document manager should have all active documents.

Comment (amunger) [reply src/client/datascience/editor-integration/codelensprovider.ts:166]: I'm not saying it does happen, more wondering if this fallback is ever hit.

**Comment by IanMatthewHuff** [comment]
Is this just running logging on CI currently? If so you can always create it as a draft until it's ready for review.

**Comment by amunger** [comment]
I haven't seen the same test fail on this branch, so I'm planning on just checking this in to help troubleshoot if it becomes flakey in the future

**Review by rchiodo — APPROVED** [review]

**Pull request merged**